xclim/indices/_agro.py

```diff
@@ -1418,3 +1419,75 @@ def effective_growing_degree_days(
     egdd = aggregate_between_dates(deg_days, start=start, end=end, freq=freq)
 
     return to_agg_units(egdd, tas, op="delta_prod")
+
+
+@declare_units(tasmin="[temperature]")
+def hardiness_zones(
+    tasmin: xarray.DataArray, window: int = 30, method: str = "usda", freq: str = "YS"
+):
+    """Hardiness zones.
+
+    Hardiness zones are a categorization of the annual extreme temperature minima, averaged over a certain period.
+    The USDA defines 14 zones, each divided into two sub-zones, using steps of 5°F, starting at -65°F.
+    The Australian National Botanic Gardens define 7 zones, using steps of 5°C, starting at -15°C.
+
+    Parameters
+    ----------
+    tasmin : xr.DataArray
+        Minimum temperature.
+    window : int
+        The length of the averaging window, in years.
+    method : {'usda', 'anbg'}
+        Whether to return the American (`usda`) or the Australian (`anbg`) classification zones.
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xr.DataArray, [dimensionless]
+        {method} hardiness zones.
+        US sub-zones are denoted by using a half step. For example, Zone 4b is given as 4.5.
+        Values are given at the end of the averaging window.
+
+    References
+    ----------
+    :cite:cts:`usda_2012`
+    :cite:cts:`dawson_plant_1991`
+    """
+    # Construct climatology
+    if parse_offset(freq) != (1, "A", True, "JAN"):
+        raise ValueError(f"Freq not allowed: {freq}. Must be `YS` or `AS-JAN`")
```

Comment:
My understanding is that these checks are done at the indicator level using the `allowed_period` attribute.

Reply:
Yes. I believe the issue is that if `freq` is not also present in the indice, we can't add `allowed_freq` to the wrapped indicator. This is also relevant for the indice as well (hardiness at the monthly timestep makes no sense).

I copied this from elsewhere, but something I'd like to confirm is if this prevents us from using the different annual start dates (`AS-JUL`). This should be allowed.

Reply:
Indicator only checks if the base (here "A") is a match. 

See `ResamplingIndicator._preprocess_and_checks`

Reply:
`xc.core.calendar.parse_offset("AS-JUL")` yields `(1, 'A', True, 'JUL')`, so you will indeed raise a ValueError and "AS-JUL" won't be permitted as things are. If you just want yearly frequency, maybe you could use:
```python
_, base, _, _ = parse_offset(freq)
if base != "A":
    ...
```

Reply:
This is covered in the wrapped indicator. We need to trust that users won't compute something that makes no sense with the index.